### PR TITLE
Remove sub-path from container registry realm

### PR DESCRIPTION
**Pull request by wxiaoguang**
Container registry requires that the "/v2" must be in the root, so the sub-path in AppURL should be removed

### CODE REVIEW

**Comment by KN4CK3R** [comment]
Could you remove https://github.com/go-gitea/gitea/blob/0188d82e4908eb173f7203d577f801f3168ffcb8/modules/setting/packages.go#L69-L70 too? That should use the new logic too (maybe with a new `GuessCurrentHost(ctx)` method?).

**Comment by wxiaoguang** [comment]
It seems that `RegistryHost` is for display-only purpose. So I guess we can leave it as-is? To remove the `RegistryHost`, it needs to touch much more code, at the moment I don't have enough time to fully test them ....

image

**Comment by KN4CK3R** [comment]
Fine for me 👍

File: tests/integration/api_packages_container_test.go [hunk 1]
```diff
@@ -84,15 +84,17 @@ func TestPackageContainer(t *testing.T) {
 			Token string `json:"token"`
 		}
 
-		authenticate := []string{`Bearer realm="` + setting.AppURL + `v2/token",service="container_registry",scope="*"`}
+		wwwAuthenticateValues := func() []string {
+			return []string{`Bearer realm="` + setting.AppURL + `v2/token",service="container_registry",scope="*"`}
+		}
```
Comment (KN4CK3R) [inline tests/integration/api_packages_container_test.go]: Why a func?

Comment (wxiaoguang) [reply tests/integration/api_packages_container_test.go]: Because the AppURL might change during the tests. So it should always read the current AppURL

**Comment by wxiaoguang** [comment]
> Fine for me 👍

Managed to remove the RegistryHost in a62ec02a901710e310187b9cd47ad2a2b13721f2  (not really tested ....)

File: tests/integration/api_packages_container_test.go (hunk 1 above)
Comment (wxiaoguang) [reply tests/integration/api_packages_container_test.go]: Read the code again, now we don't need this function. So just rename it to `defaultAuthenticateValues`

**Review by KN4CK3R — APPROVED** [review]

File: tests/integration/api_packages_container_test.go (hunk 1 above)
Comment (KN4CK3R) [reply tests/integration/api_packages_container_test.go]: Yep, approved already 👍

**Review by lunny — APPROVED** [review]

**Pull request merged**